### PR TITLE
implementing TransactionalQueue addAll method

### PR DESCRIPTION
**Pull request by alex-dukhno**
Creating WIP PR for cross referencing with https://github.com/hazelcast/hazelcast-client-protocol/pull/269

### CODE REVIEW

**Comment by alex-dukhno** [comment]
The PR adds `addAll(Collection<? super E> collection)` and `addAll(E... items)` (from #4233) for `TransactionalQueue`

@noctarius could you please take a look?

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    public TransactionalQueueAddAllMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Object innerCall() throws Exception {
+        final TransactionContext context = endpoint.getTransactionContext(parameters.txnId);
+        final TransactionalQueue queue = context.getQueue(parameters.name);
+        return queue.addAll(parameters.items);
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java]: how do we utilize `timeout`?

Comment (alex-dukhno) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java]: Thank you for noticing this. I've addressed it.

**Comment by alex-dukhno** [comment]
@ihsandemir I've regenerated codecs and rebased over current master.

Thank you for your review

**Comment by mmedenjak** [comment]
@ihsandemir are all review comments addressed? Can we merge it now?

**Comment by mmedenjak** [comment]
@alex-dukhno can you rebase?

**Comment by alex-dukhno** [comment]
@mmedenjak, code is rebased

**Comment by alex-dukhno** [comment]
@ihsandemir I've rebased the code and addressed https://github.com/hazelcast/hazelcast/pull/16153#discussion_r362741442

**Comment by mdumandag** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.QueuePermission;
+import com.hazelcast.transaction.TransactionContext;
+import com.hazelcast.transaction.TransactionalQueue;
+
+import java.security.Permission;
+import java.util.concurrent.TimeUnit;
+
+public class TransactionalQueueAddAllMessageTask
```
Comment (alex-dukhno) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java]: @mmedenjak @ihsandemir I see that `TransactionalQueue*MessageTask`s are stored into the map in `DefaultMessageTaskFactoryProvider`, but it looks like is generated or something.
Test in the PR (and similar PRs for other operations) is passed, but I am worried that it won't work in real scenario if `TransactionalQueueAddAllMessageTask` won't be stored in `DefaultMessageTaskFactoryProvider`.

Do you guys know what is need to be done? Thank you

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java]: You can edit `DefaultMessageTaskFactoryProvider`, the comment at the top is not correct. This file not generated.

Comment (alex-dukhno) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/transactionalqueue/TransactionalQueueAddAllMessageTask.java]: Ok, 🤦‍♂ 😄 

Thank you

**Comment by mmedenjak** [comment]
Moving into 4.1 as we have reached the deadline for 4.0. Don't worry, we'll keep reviewing and get it merged soon. Thank you for the effort.

**Comment by pveentjer** [comment]
Closing PR due to lack of progress and we should not add new functionality to our TX system.

**Pull request closed without merging**